File: android/lca/src/main/java/com/dreampany/lca/ui/model/ExchangeItem.java
/*
package com.dreampany.lca.ui.model;

import androidx.annotation.LayoutRes;
import androidx.annotation.NonNull;
import androidx.annotation.StringRes;
import android.view.View;

import com.dreampany.frame.ui.model.BaseItem;
import com.dreampany.lca.R;
import com.dreampany.lca.data.model.Exchange;

import java.io.Serializable;
import java.util.List;

import eu.davidea.flexibleadapter.FlexibleAdapter;
import eu.davidea.flexibleadapter.items.IFlexible;

*/
/**
 * Created by Hawladar Roman on 5/31/2018.
 * BJIT Group
 * hawladar.roman@bjitgroup.com
 *//*

public class ExchangeItem extends BaseItem<Exchange, ExchangeItem.ViewHolder> {

    private ExchangeItem(@NonNull Exchange exchange,
                         @LayoutRes int layoutId) {
        super(exchange, layoutId);
    }

    public static ExchangeItem getItem(@NonNull Exchange exchange) {
        return new ExchangeItem(exchange, 0);
    }

    @Override
    public ViewHolder createViewHolder(View view, FlexibleAdapter<IFlexible> adapter) {
        return new ViewHolder(view, adapter);
    }

    @Override
    public void bindViewHolder(FlexibleAdapter<IFlexible> adapter, ViewHolder holder, int position, List<Object> payloads) {

    }

    @Override
    public boolean filter(Serializable constraint) {
        return false;
    }

    static class ViewHolder extends BaseItem.ViewHolder {

        protected ViewHolder(@NonNull View view, @NonNull FlexibleAdapter adapter) {
            super(view, adapter);
        }

        String getText(@StringRes int keyResId) {
            return getContext().getString(keyResId);
        }

        String getItemText(@StringRes int keyResId, String value) {
            return String.format(
                    getText(R.string.coin_format),
                    getText(keyResId),
                    value);
        }

        String getItemText(@StringRes int keyResId, double value) {
            return String.format(
                    getText(R.string.coin_format),
                    getText(keyResId),
                    value);
        }
    }


}
*/

File: android/lca/src/main/kotlin/com/dreampany/lca/ui/model/ExchangeItem.kt
package com.dreampany.lca.ui.model

import android.view.View
import androidx.annotation.LayoutRes
import androidx.recyclerview.widget.RecyclerView
import com.dreampany.framework.ui.model.BaseItem
import com.dreampany.lca.data.model.Exchange
import com.dreampany.lca.misc.Constants
import eu.davidea.flexibleadapter.FlexibleAdapter
import eu.davidea.flexibleadapter.items.IFlexible
import java.io.Serializable

/**
 * Created by roman on 2019-08-03
 * Copyright (c) 2019 bjit. All rights reserved.
 * hawladar.roman@bjitgroup.com
 * Last modified $file.lastModified
 */
class ExchangeItem private constructor(
    item: Exchange, @LayoutRes layoutId: Int = Constants.Default.INT

) : BaseItem<Exchange, ExchangeItem.ViewHolder, String>(item, layoutId) {

    companion object {
        fun getItem(exchange: Exchange): ExchangeItem {
            return ExchangeItem(exchange, 0);
        }
    }

    override fun createViewHolder(
        view: View,
        adapter: FlexibleAdapter<IFlexible<RecyclerView.ViewHolder>>
    ): ViewHolder {
        return ViewHolder(view, adapter)
    }

    override fun filter(constraint: String): Boolean {
        return false
    }

    class ViewHolder(
        view: View,
        adapter: FlexibleAdapter<*>
    ) : BaseItem.ViewHolder(view, adapter) {


        init {
        }

        override fun <VH : BaseItem.ViewHolder, T : Base, S : Serializable, I : BaseItem<T, VH, S>>
                bind(position: Int, item: I) {
        }
    }
}